assets/src/stories-editor/blocks/amp-story-page/edit.js

```diff
@@ -106,15 +119,51 @@ class PageEdit extends Component {
 
 			mediaType = media.type;
 		}
+		const mediaUrl = has( media, [ 'sizes', MAX_IMAGE_SIZE_SLUG, 'url' ] ) ? media.sizes[ MAX_IMAGE_SIZE_SLUG ].url : media.url;
+		const isExcessiveVideoSize = VIDEO_BACKGROUND_TYPE === mediaType && this.isVideoSizeExcessive( media.filesizeInBytes, media.fileLength );
 
 		this.props.setAttributes( {
-			mediaUrl: media.url,
+			mediaUrl,
 			mediaId: media.id,
 			mediaType,
+			isExcessiveVideoSize,
 			poster: VIDEO_BACKGROUND_TYPE === mediaType && media.image && media.image.src !== media.icon ? media.image.src : undefined,
 		} );
 	}
 
+	/**
+	 * Gets whether the file is more than 1 MB per second, like 3 MB for 3 seconds.
+	 *
+	 * @param {number} fileSize The size of the file in bytes.
+	 * @param {string} length A colon-separated time length of the file, like '01:04'.
+	 * @return {boolean} Whether the file size is more than 1MB per second.
+	 */
+	isVideoSizeExcessive( fileSize, length ) {
+		const fileLengthInSeconds = this.getSecondsFromTime( length );
+		return fileSize > fileLengthInSeconds * MEGABYTE_IN_BYTES;
+	}
+
+	/**
+	 * Gets the number of seconds in a colon-separated time string, like '01:10'.
+	 *
+	 * @param {string} time A colon-separated time, like '0:12'.
+	 * @return {number} seconds The number of seconds in the time, like 12.
+	 */
+	getSecondsFromTime( time ) {
```

Comment:
Minor point, but since `getSecondsFromTime` and `isVideoSizeExcessive` are not really tied to this class, they could be moved to the helpers file where we could also more easily write unit tests for them. Alternatively, the methods could be declared static and also have tests more easily.

Reply:
Good point, https://github.com/ampproject/amp-wp/pull/2509/commits/82b827436ee0bbe723ebf9b5d8f8b9804f462cfb moves these functions to `helpers/index.js`.